src/main/java/com/laytonsmith/core/events/drivers/EntityEvents.java

```diff
@@ -2053,6 +2055,91 @@ public Driver driver() {
 		}
 	}
 
+	@api
+	public static class entity_toggle_swim extends AbstractGenericEvent<MCEntityToggleSwimEvent> {
+
+		@Override
+		public String getName() {
+			return "entity_toggle_swim";
+		}
+
+		@Override
+		public String docs() {
+			return "{type: <macro> The entity type of the entity | id: <macro> The entity id of the entity"
+					+ " | player: <macro> The player triggering the event}"
+					+ " This event is called when an entity's swimming status is toggled"
+					+ " {id: The entityID of the entity | type: The entity type of the entity |"
+					+ " swimming: true if the entity is entering swimming mode, false if the entity is leaving it |"
+					+ " player: If the entity is a player, this will contain their name, otherwise null}" + " {}"
+					+ " {}";
+		}
+
+		@Override
+		public boolean matches(Map<String, Mixed> filter, MCEntityToggleSwimEvent e) throws PrefilterNonMatchException {
```

Comment:
Instead of matches(), override the getPrefilterBuilder() method. This new method is much faster at runtime, and offers additional benefits such as better documentation. There are several examples of this, just search for "getPrefilterBuilder" to see some examples you can learn from (player_chat for instance has been converted), or see the PrefilterBuilder class. Once you implement that method, the matches() method goes away.

Reply:
Changed it, but for a player that may not exist and the id of the entity didn't appear to have any examples (or I didn't notice). For the new filter I used the documentation from a similar one, as I don't quite understand how their tags work.